pkg/apis/hive/v1alpha1/hiveconfig_types.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HiveConfigSpec defines the desired state of Hive
+type HiveConfigSpec struct {
+	// Image controls the image used for the Hive controllers, as well
+	// as deprovision pods. The default if left empty is to use latest master.
+	Image string `json:"image,omitempty"`
+}
+
+// HiveConfigStatus defines the observed state of Hive
+type HiveConfigStatus struct {
```

Comment:
I'm not sure if we can do anything useful with a generation, or which generation we'd record as there are a few in play, but if there was any use we certainly could.

I didn't have any plans for ready or not but we do have cards for health checks so this might be an ok place.

Reply:
Could we embed the openshift/api's operator structs for spec/status even if we don't use most of the fields at the moment? 
https://github.com/openshift/api/blob/master/operator/v1/types.go#L43-L64
https://github.com/openshift/api/blob/master/operator/v1/types.go#L113-L129

Reply:
Will do!